File: ios/Sources/ElapsedRealtimePlugin/ElapsedRealtimePlugin.swift
import Foundation
import Capacitor

/**
 * Please read the Capacitor iOS Plugin Development Guide
 * here: https://capacitorjs.com/docs/plugins/ios
 */
@objc(ElapsedRealtimePlugin)
public class ElapsedRealtimePlugin: CAPPlugin, CAPBridgedPlugin {
    public let identifier = "ElapsedRealtimePlugin"
    public let jsName = "ElapsedRealtime"
    public let pluginMethods: [CAPPluginMethod] = [
        CAPPluginMethod(name: "getElapsedRealtime", returnType: CAPPluginReturnPromise)
    ]
    private let implementation = ElapsedRealtime()

    @objc func getElapsedRealtime(_ call: CAPPluginCall) {
        let elapsedTime = ProcessInfo.processInfo.systemUptime * 1000 // Convert to milliseconds
        call.resolve([
            "elapsedTime": elapsedTime
        ])
    }
}

File: android/src/main/java/br/com/risoluto/elapsedrealtime/ElapsedRealtimePlugin.java
package br.com.risoluto.elapsedrealtime;

import android.os.SystemClock;

import com.getcapacitor.JSObject;
import com.getcapacitor.Plugin;
import com.getcapacitor.PluginCall;
import com.getcapacitor.PluginMethod;
import com.getcapacitor.annotation.CapacitorPlugin;

@CapacitorPlugin(name = "ElapsedRealtime")
public class ElapsedRealtimePlugin extends Plugin {

    private ElapsedRealtime implementation = new ElapsedRealtime();

    @PluginMethod
    public void echo(PluginCall call) {
        String value = call.getString("value");

        JSObject ret = new JSObject();
        ret.put("value", implementation.echo(value));
        call.resolve(ret);
    }

    @PluginMethod
    public void getElapsedRealtime(PluginCall call) {
        long elapsedTime = SystemClock.elapsedRealtime();
        JSObject ret = new JSObject();
        ret.put("elapsedTime", elapsedTime);
        call.resolve(ret);
    }
}
